### ISSUE TITLE
Can mpdecimate_trim create smaller sized videos that are the same length of time as the original video in order to  retain all of the original audio?

### ISSUE THREAD

**Issue by Yekutiel** [KenjiTakahashi/mpdecimate_trim#5]
I presume the following is far beyond the scope of this project, but I figured I'd ask just in case I am wrong.
 
> Script to trim similar/duplicate fragments from video clips. While keeping audio in sync.
 
Does that mean only the audio associated with portions of the video that are **not** trimmed will be included in the video clips which `mpdecimate_trim` creates? 
 
I intend to use OBS on Linux to create screencasts of my coding sessions which I will narrate. In other words, I intend to talk while I code. As a result, when I review my coding sessions I will be able to quickly be reminded that, for example, "Oh,  yeah, that's what I was thinking when I wrote that line of code."
 
I presume that sometimes I will stop coding, yet continue narrating. Normally, in such cases, nothing would change on my monitor because I would neither be using my keyboard nor my mouse. Would `mpdecimate_trim` delete my narration (the audio) associated with the frames it had “mpdecimated”?
 
I don't want to lose any of my narration.
 
Is there a way to decrease the file size of my screencasts—say, by using `mpdecimate_trim`—yet not actually decrease the length of time of my screencasts so that all of my narration (the audio) would be synched up with the video, which would be precisely the same length as it was before `mpdecimate_trim` was used?
 
I know next to nothing about video files, but I  assume my feature request would require the video player to be able to pause the video portion of a file—yet continue the audio portion of a file—at those points where `mpdecimate_trim` had removed duplicate video frames. 
 
In other words, if a sequence of frames were frame A, frame B, frame C, and frame D and if all four frames were identical, and if `mpdecimate_trim` had removed frame B, frame C, and frame D, then I assume the video player would need to pause frame A so that sequence of frames consisting of frame A, frame B, frame C, and frame D would be replaced by sequence of frames consisting of frame A played for the first frame and then paused for the second, third, and fourth frames, yet play the **audio** that had been associated with frame B, frame C, and frame D.
 
My hunch is that this is not a feature which normal video players support. I use MPlayer which, as you might know, has a remarkable number of features. But I kind of doubt even “Mighty MPlayer” has that sort of ability.

**Comment by KenjiTakahashi** [comment]
You're right that this tool is not for you. It is explicitly meant to remove frames completely, along with the audio parts.
It sounds that you're looking to reduce the file size for videos that have "still portions" in them.
It is not possible the way you described, but it is actually what "frame dropping" mentioned in the README does. It finds identical (or "similar enough", if a decimation algorithm is used) frames and replaces duplicates with pointers to the original one.
Pretty sure there is a `ffmpeg` incantation that will do it for you, but I'm afraid you'll have to research the details for yourself.

**Comment by Yekutiel** [comment]
Thanks for taking the time to carefully respond to me.

> It sounds that you're looking to reduce the file size for videos that have "still portions" in them.

That is correct.

To be clear, let's say I were to use OBS studio to create a screencast for 60 minutes while I was coding. Furthermore, let's say that for 10 to 15 minutes during that screencast I were to stop using my mouse and keyboard, yet continue talking as I, for example, thought out loud. For example, I might stop using my mouse and keyboard, yet say, "Hmmm. I'm a little confused here. I wonder if I should..."

During such times I would want my voice recorded; however, because I would neither be using my mouse nor my keyboard,  software (computer code) that, for example, replaced video frames that were identical (or "similar enough", if a decimation algorithm were used) with pointers to the original video frame could significantly reduce the file size.

> Pretty sure there is a `ffmpeg` incantation that will do it for you, but I'm afraid you'll have to research the details for yourself.

Given the extreme power and popularity of `ffmpeg` along with the usefulness of such a feature to many users, I presumed that  `ffmpeg` must have such a feature.

**Comment by aaronse** [comment]
@Yekutiel, curious, did you find a solution that keeps synchronized visual frames and audio samples for segments where audio activity is detected?

**Comment by Yekutiel** [comment]
I did not.
